services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/outputs/_directory_utils.py

```diff
@@ -0,0 +1,21 @@
+import os
+from pathlib import Path
+
+from pydantic import ByteSize, parse_obj_as
+
+
+def get_directory_total_size(path: Path) -> ByteSize:
+    """return the size of a directory in bytes"""
+    # NOTE: performance of this degrades with large amounts of files
+    # until we do not hit 1 million it can be ignored
+    # NOTE: file size has no impact on performance
+    if not path.exists():
+        return parse_obj_as(ByteSize, 0)
+
+    total = 0
+    for entry in os.scandir(path):
+        if entry.is_file():
+            total += entry.stat().st_size
+        elif entry.is_dir():
+            total += get_directory_total_size(Path(entry.path))
+    return parse_obj_as(ByteSize, total)
```

Comment:
actually the order of magnitude slower you are talking about probably comes from pydantic, not from using pathlib

Reply:
I also thought about that but apparently this has no noticeable impact. I have benchmarked your previous suggestion without any pydantic parsing and the result was the same.
I've added pydantic here and it had no noticeable impact.

If you think about it it would make sense. The `scandir` operation operates on memory (the disk) that is one order of magnitude slower or worse.
The effects of pydantic are irrelevant in this situation.